src/main/java/executor/service/service/parallelflowexecutor/impls/subscribers/ExecutableScenarioComposerImpl.java

```diff
@@ -0,0 +1,90 @@
+package executor.service.service.parallelflowexecutor.impls.subscribers;
+
+import executor.service.model.proxy.ProxyConfigHolder;
+import executor.service.model.scenario.Scenario;
+import executor.service.model.service.ExecutableScenario;
+import executor.service.queue.ProxyQueue;
+import executor.service.queue.ScenarioQueue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExecutableScenarioComposerImpl implements ExecutableScenarioComposer {
+
+    private static final Logger log = LoggerFactory.getLogger(ExecutableScenarioComposerImpl.class);
+
+    private final ProxyQueue proxyQueue;
+    private final ScenarioQueue scenarioQueue;
+    private final ProxyConfigHolder defaultProxy;
+
+    public ExecutableScenarioComposerImpl(ProxyQueue proxyQueue, ScenarioQueue scenarioQueue, ProxyConfigHolder defaultProxy) {
+        this.proxyQueue = proxyQueue;
+        this.scenarioQueue = scenarioQueue;
+        this.defaultProxy = defaultProxy;
+    }
+
+    @Override
+    public List<ExecutableScenario> composeExecutableScenarios() {
+        try {
+            List<ProxyConfigHolder> proxies = proxyQueue.getAllProxy();
+            List<Scenario> scenarios = scenarioQueue.getAllScenario();
+            if (!scenarios.isEmpty() && !proxies.isEmpty()) {
```

Comment:
Here you can use the proxyQueue size check, adding logic to the ProxyQueue to determine the size, which will make it possible not to do **proxies.forEach(proxyQueue::addProxy);** to avoid unnecessary resource consumption.